jep/302/README.adoc

```diff
@@ -122,37 +125,81 @@ We need to finish up the work on the BOM to be more precise here.
+////
 
-This subpart especially is welcoming feedback.
+Here is what it will look like:
+
+```
+/var/jenkins/
+├── home
+│   ├── jobs
+│   │   └── the_job # configuration file only
+│   ├── nodes
+│   ├── plugins
+│   ├── secrets
+│   ├── updates
+│   ├── userContent
+│   └── users
+└── var
+    ├── logs # JENKINS-50291
+    │   └── tasks
+    ├── plugins # exploded plugins, using --pluginroot switch
+    ├── jobs # JENKINS-50164
+    │   └── the_job
+    │       ├── builds
+    │       └── workspace
+    └── war # using --webroot
+        ├── META-INF
+        ├── WEB-INF
+        ├── ...
+```
+
+This way, everything related to jobs, but is less important than its configuration itself is separated under a different `${JENKINS_HOME}/var/` root.
+
+Though we will for now back up these directories too, having this distinction baked in from the ground up could prove useful in the future if we need to make the snapshotting and rollback processes smarter and more robust.
 
-THIS IS A VERY VERY BAD IDEA BECAUSE [...] I'm very interested in :-).
-////
+For instance, in a case where Jenkins would not come back online after an automated rollback, we would be able to leverage this data separation to try a degraded rollback strategy where we would handle that `$JENKINS_HOME/var` directory in a different way than `JENKINS_HOME/jobs` or other ones.
 
-We plan to configure Jenkins to use non default values to put the workspace and builds not under the same location as job configuration itself.
+===== What to back up
 
-_Workspace Root Directory_:: `${JENKINS_HOME}/workspace/${ITEM_FULLNAME}`
-_Build Record Root Directory_::
-`${JENKINS_HOME}/var/${ITEM_FULL_NAME}/builds`
+Thanks to the data segregation explained above, we will be snapshotting (almost) everything under `/var/jenkins/home`.
 
-Though we will for now backup these directories too, having this distinction baked in from the ground up could prove useful in the future if we need to make the snapshotting and rollback processes smarter and more robust.
+We still need to have a `.gitignore` file for some things that either cannot be moved elsewhere, or that we do not want to store in the Git repository.
+As said above, this will likely be improved as we go.
 
-For instance, in a case where Jenkins would not come back online after an automated rollback, we would be able to leverage this data separation to try a degraded rollback strategy where we would handle that `$JENKINS_HOME/var` directory in a different way than `JENKINS_HOME/jobs` or other ones.
+[source,gitignore,title=.gitignore]
+----
+plugins/
+secrets/master.key
+----
+
+====== about `/var/jenkins/home/plugins`
+
+This directory contains the hpi/jpi files before extraction.
+Ideally, we should be moving this elsewhere under `/var/jenkins/var/plugins`, but it's not doable yet currently (`--pluginsroot` only configures a different location for exploded plugins).
```

Comment:
Careful here. Currently `$JENKINS_HOME/plugins/shortName.jpi` is the definitive record that a plugin was installed (there is also a `shortName.jpi.disabled` file if disabled), and in what version. If you do not back that up, then you are no longer versioning the Jenkins configuration.

Now I understand that for plugins in Essentials, this is exactly right—the Essentials manifest is in control of what plugins are there. (Again, excepting `*.jpi.disabled` files, so I think what you meant to exclude is more narrowly `plugins/*.jpi`.) But what about “inessential” plugins? There will be no trace in the Git history of their presence or version.

Admittedly a rollback is not going to _need_ to revert these, either, since the Essentials upgrade would not have touched them. Just seems a little risky. People may be expecting that a Git revision (plus `master.key`) corresponds to a fully reconstructible Jenkins instance.

Reply:
Maybe the `.gitignore` can simply list those plugins actually in Essentials?

```
plugins/structs.jpi
# …
```

Reply:
> People may be expecting that a Git revision (plus master.key) corresponds to a fully reconstructible Jenkins instance.

Well, then they will be deceived... It's not a backup system.

To me that this part is going to be out of the scope of the snapshotting system, exactly for the reason you gave: I consider that the whole plugin list is defined explicitly by the BOM, for Essentials plugin.

I would say that for others, well it's also out of scope, and it will be the evergreen client responsibility to store the versions of **all** plugins present there, to be able to restore the previous version if we rollback.